app/src/main/java/com/example/fitjournal/home/presentation/components/card/WeightLiftingCard.kt

```diff
@@ -41,11 +39,20 @@ fun WeightLiftingCard(
                 isWeightTrainingIcon = true
             )
             Spacer(modifier = Modifier.height(Spacing.spacing8))
-            TopSetTitle()
+            if (weightLiftingUi.sets != null &&
+                weightLiftingUi.reps != null &&
+                weightLiftingUi.weight != null
+            ) {
```

Comment:
I'm seeing these null checks a lot. I think it'd be better if we extract this out into a function within the WeightLiftingUi data class.

```suggestion
           data class WeightLiftingUi( 
           // some properties
           ){
               val isNotNull get() = sets != null && reps != null && weight != null
           }
```